tools/copyblocks/bucket.go

```diff
@@ -0,0 +1,232 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package main
+
+import (
+	"context"
+	"io"
+	"strings"
+	"time"
+
+	"cloud.google.com/go/storage"
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob"
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob/blob"
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob/container"
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob/sas"
+	"github.com/grafana/dskit/backoff"
+	"github.com/pkg/errors"
+	"google.golang.org/api/iterator"
+)
+
+type bucket interface {
+	Get(ctx context.Context, name string) (io.ReadCloser, error)
+	Copy(ctx context.Context, name string, dstBucket bucket) error
+	ListPrefix(ctx context.Context, prefix string, recursive bool) ([]string, error)
+	UploadMarkerFile(ctx context.Context, name string) error
+	Name() string
```

Comment:
[nit] Is there another name we can use for the `name` parameters to make it clear that they're referring to object names rather than the bucket name?

Reply:
swapped to `objectName`